solr/core/src/java/org/apache/solr/servlet/RateLimitManager.java

```diff
@@ -147,19 +145,16 @@ public boolean handleRequest(HttpServletRequest request) throws InterruptedExcep
 
   // Decrement the active requests in the rate limiter for the corresponding request type.
   public void decrementActiveRequests(HttpServletRequest request) {
-    RequestRateLimiter.AcquiredSlotMetadata acquiredSlotMetadata = activeRequestsMap.get(request);
+    RequestRateLimiter.SlotMetadata slotMetadata = activeRequestsMap.get(request);
 
-    if (acquiredSlotMetadata == null) {
+    if (slotMetadata == null) {
```

Comment:
I think I would flip this condition so that we only have one return point from the method.